Add new inlineEncodeASCIISymbol

dchopra001 opened the pull request:
Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

dchopra001 commented:
See https://github.com/eclipse-openj9/openj9/pull/13855/ for related PR

jdmpapin commented on compiler/compile/OMRSymbolReferenceTable.cpp:
```diff
@@ -676,6 +676,19 @@ OMR::SymbolReferenceTable::findOrCreateCurrentTimeMaxPrecisionSymbol()
    }
 
+TR::SymbolReference *
+OMR::SymbolReferenceTable::findOrCreateEncodeASCIISymbol()
```
`encodeASCIISymbol` does need to be defined in OMR (unfortunately), but this method could be defined in OpenJ9 to reduce the amount of OpenJ9-specific code being added here

jdmpapin commented on compiler/compile/OMRSymbolReferenceTable.cpp:
```diff
@@ -676,6 +676,19 @@ OMR::SymbolReferenceTable::findOrCreateCurrentTimeMaxPrecisionSymbol()
+TR::SymbolReference *
+OMR::SymbolReferenceTable::findOrCreateEncodeASCIISymbol()
+   {
+   if (!element(encodeASCIISymbol))
+      {
+      TR::MethodSymbol * sym = TR::MethodSymbol::create(trHeapMemory(),TR_Helper);
+      sym->setHelper();
+
+      element(encodeASCIISymbol) = new (trHeapMemory()) TR::SymbolReference(self(), encodeASCIISymbol, sym);
+      }
+   return element(encodeASCIISymbol);
```
It would probably be good to use `findOrCreateCodeGenInlinedHelper()` for this. See `J9::SymbolReferenceTable::findOrCreateNonNullableArrayNullStoreCheckSymbolRef()` for an example

jdmpapin commented on compiler/ras/Debug.cpp:
```diff
@@ -1601,6 +1601,8 @@ TR_Debug::getName(TR::SymbolReference * symRef)
          case TR::SymbolReferenceTable::currentTimeMaxPrecisionSymbol:
             return "<currentTimeMaxPrecision>";
+         case TR::SymbolReferenceTable::encodeASCIISymbol:
+            return "<encodeASCIISymbol>";
```
Not a big deal at all, but the name here and in the table below can omit the `Symbol` suffix like most of the other "non-helper" symbols do

jdmpapin commented on compiler/compile/OMRSymbolReferenceTable.cpp:
```diff
@@ -676,8 +676,6 @@ OMR::SymbolReferenceTable::findOrCreateCurrentTimeMaxPrecisionSymbol()
    }
 
-
-
```
Stray whitespace change

jdmpapin reviewed (approved):
LGTM with one nit pick

jdmpapin commented:
Jenkins build all

dchopra001 commented:
> By the way, please squash

Squashed and removed the whitespace [change](https://github.com/eclipse/omr/pull/6313#discussion_r817887582) in the final commit.
